src/request.c

```
@@ -717,19 +709,19 @@ node_create(const char *msg, const char *apikey)
 		goto cleanup;
 	}
 
+	dao_network_get_network(&network_uid, &subnet, &netmask, client_id, network_description);
 
-	if (dao_network_get_ippool(network_description, &network_uid,
-	    &subnet, &netmask, &tmp_pool) < 0) {
-		log_warnx("%s: dao_network_get_ippool", __func__);
-		goto cleanup;
-	}
+	/* Try to find an address available */
+	dao_ipv4_get_available(&ipaddress, network_uid);
 
-	ippool = ippool_new(subnet, netmask);
-	free(ippool->pool);
-	ippool->pool = (uint8_t*)tmp_pool;
-	tmp_pool = NULL;
-	pool_size = (ippool->hosts+7)/8 * sizeof(uint8_t);
-	ipaddress = ippool_get_ip(ippool);
+	/* We found no address available, find the last one allocated */
+	if (ipaddress == NULL) {
+		dao_ipv4_get_last(&last_alloc_addr, network_uid);
+
+		/* If last_alloc_addr is NULL, it will start from the subnet address */
+		if ((ipaddress = get_next_address(subnet, netmask, last_alloc_addr)) == NULL)
```

So I am a tad confused about this case and the comment. If the ipaddress returned is NULL, i'm assuming this is the case where everything in the subnet has been allocated. Would it make sense to log this (and eventually set up an alert in papertrail) rather than just silently going to cleanup?

---

I agree with that. I'll add a logging here. I guess because I wasn't sure how I was going to handle the errors inside get_next_address() I wasn't sure where I wanted to report about the subnet being exhausted. From inside the function or here outside. Anyway adding a log line, and will check later where I want to go with that